src/components/Dashboard/UrlListItem.tsx

```diff
@@ -11,102 +9,69 @@ import { deleteUrlApi } from '@/services/api';
 import { UrlType } from '@/types/url.types';
 import formatDate from '@/utils/formatDate';
 
-import Button from '../Button';
-import { Loader } from '../Loader';
+import { CopyButton, DeleteButton } from './Buttons';
 
 interface UrlListItemProps {
     url: UrlType;
     copyButtonHandler: (url: string) => void;
 }
-
-const UrlListItem = ({ url, copyButtonHandler }: UrlListItemProps) => {
+const UrlListItem: React.FC<UrlListItemProps> = ({ url, copyButtonHandler }) => {
+    const { userData } = useAuthenticated();
     const deleteMutation = useMutation({
         mutationFn: deleteUrlApi,
-        onSuccess: () => {
-            queryClient.invalidateQueries(['urls']);
-        },
+        onSuccess: () => queryClient.invalidateQueries(['urls']),
         onError: (error) => {
             window.alert('Error deleting URL');
             console.error(error);
         },
     });
-    const { userData } = useAuthenticated();
 
-    return (
-        <li className="rounded-lg grid grid-cols-1 sm:grid-cols-[1fr,32px] bg-white w-full px-2 py-2 sm:px-4 sm:py-2 items-center flex-wrap">
-            <div className="flex items-center gap-2">
-                <div className="mr-2 flex items-center justify-center">
-                    <TbWorldWww className="text-black text-opacity-70 text-4xl" />
-                </div>
+    const handleCopy = () => copyButtonHandler(`${TINY_SITE}/${url.shortUrl}`);
+    const handleDelete = () => deleteMutation.mutate({ id: url.id, userId: Number(userData?.data?.id) });
 
-                <div className="flex flex-col w-[85%]">
-                    <div className="flex max-w-fit items-center space-x-2 ">
-                        <Link
-                            href={`${TINY_SITE}/${url.shortUrl}`}
-                            target="_blank"
-                            rel="noreferrer"
-                            className="w-full  truncate text-sm font-semibold text-blue-800 sm:max-w-[300px] sm:text-base md:max-w-[360px] xl:max-w-[500px]"
-                        >
-                            {TINY_SITE}/{url.shortUrl}
-                        </Link>
-                        <Button
-                            className="rounded-full bg-gray-100 p-1.5 transition-all duration-75 hover:scale-105 hover:bg-blue-100 active:scale-95"
-                            onClick={() => copyButtonHandler(`${TINY_SITE}/${url.shortUrl}`)}
-                            testId="copy-button"
-                        >
-                            <span className="sr-only">Copy</span>
-                            <MdOutlineContentCopy className="text-black" />
-                        </Button>
-                    </div>
+    return (
+        <li className="rounded-xl bg-white px-2 py-2 sm:pl-4 sm:py-4 flex flex-col overflow-hidden">
```

Comment:
- if we use url list anywhere else it also renders `li` in the dom, can we remove the `li` tag instead and allow the component consumer to add this tag?

Reply:
fixed